Add 'remote_parameters' and 'get_module_rref' to RemoteModule docs.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54645 Add 'remote_parameters' and 'get_module_rref' to RemoteModule docs.**

Had to replace RRef[..] with just RRef in the return signature since
sphynx seemed to completely mess up rendering RRef[..]

![Screen Shot 2021-03-24 at 5 42 34 PM](https://user-images.githubusercontent.com/9958665/112401761-75a49400-8cc8-11eb-9d8b-95ff7b2b330b.png)

Differential Revision: [D27314609](https://our.internmc.facebook.com/intern/diff/D27314609/)

pritamdamania committed:
Add 'remote_parameters' and 'get_module_rref' to RemoteModule docs.

Had to replace RRef[..] with just RRef in the return signature since
sphynx seemed to completely mess up rendering RRef[..]

Differential Revision: [D27314609](https://our.internmc.facebook.com/intern/diff/D27314609/)

[ghstack-poisoned]

SciPioneer commented on torch/distributed/nn/api/remote_module.py:
```diff
@@ -196,9 +196,11 @@ def __init__(
-        r"""Returns a list of RRefs of remote module parameters.
+    def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef]:
+        """
+        Returns a list of RRefs of remote module parameters.
```
Since we have to change `RRef[Parameter]` to `RRef` in the signature, probably we can add `Parameter` back in the comment to make up this information loss?

For example:
r"""Returns a list of RRefs of remote module parameters (`List[rpc.RRef[Parameter]]`).

SciPioneer commented on torch/distributed/nn/api/remote_module.py:
```diff
@@ -209,7 +211,7 @@ def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
         return rpc.rpc_sync(self.on, _param_rrefs, args=(self.module_rref, recurse))
 
-    def get_module_rref(self) -> rpc.RRef[nn.Module]:
+    def get_module_rref(self) -> rpc.RRef:
```
Ditto

pritamdamania committed:
Update on "Add 'remote_parameters' and 'get_module_rref' to RemoteModule docs."


Had to replace RRef[..] with just RRef in the return signature since
sphynx seemed to completely mess up rendering RRef[..]

![Screen Shot 2021-03-24 at 5 42 34 PM](https://user-images.githubusercontent.com/9958665/112401761-75a49400-8cc8-11eb-9d8b-95ff7b2b330b.png)

Differential Revision: [D27314609](https://our.internmc.facebook.com/intern/diff/D27314609/)

[ghstack-poisoned]
